Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp

```diff
@@ -845,6 +845,21 @@ static Ref<API::PageConfiguration> webkitWebViewCreatePageConfiguration(WebKitWe
         break;
     }
 
+#if ENABLE(WK_WEB_EXTENSIONS)
+    if (priv->webExtensionMode != WEBKIT_WEB_EXTENSION_MODE_NONE && webkitSettingsGetWebExtensionContext(priv->settings.get())) {
+        auto context = webkitSettingsGetWebExtensionContext(priv->settings.get());
+
+        pageConfiguration->setCrossOriginAccessControlCheckEnabled(false);
+        pageConfiguration->setProcessDisplayName(context->processDisplayName());
+        pageConfiguration->setRequiredWebExtensionBaseURL(URL(context->baseURL()));
+        pageConfiguration->setShouldRelaxThirdPartyCookieBlocking(WebCore::ShouldRelaxThirdPartyCookieBlocking::Yes);
```

Comment:
Since this is disabling security features this deserves a comment.

I'm not sure that these are long term solutions right?

Reply:
I agree, but I'd rather keep parity with Cocoa. We can investigate whether this is required on both platforms later

Reply:
Extension pages still have limited cross-origin access based on their permissions. I haven't looked where WebKit handles that, so I'm not saying there is an issue. I mostly wanted a comment to clarify why this cross-origin property was disabled to know it was considered. The Cocoa port doing so eases that worry, but still for future readers it's nice to know in case things get out of sync.